### PR TITLE
Diploma update 1.0

### PR DESCRIPTION
**Pull request by EdikFotul**

### CODE REVIEW

File: db_management/mysql/schema/initial_schema.sql [hunk 1]
```diff
@@ -1,3 +1,4 @@
+SET SQL_MODE='ALLOW_INVALID_DATES';
```
Comment (MarkHychka) [inline db_management/mysql/schema/initial_schema.sql]: Remove this

File: src/main/java/com/fitness/Gender.java [hunk 2]
```diff
@@ -4,5 +4,5 @@
  * @author Mark Hychka
  */
 public enum Gender {
-    M, F
+    M, F, SH
```
Comment (MarkHychka) [inline src/main/java/com/fitness/Gender.java]: ???

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,4 +1,4 @@
-spring.datasource.url=jdbc:mysql://localhost/fitness
-spring.datasource.username=root
-spring.datasource.password=root
+spring.datasource.url=jdbc:mysql://localhost/workout_app
```
Comment (MarkHychka) [inline src/main/resources/application.properties]: revert changes

File: src/main/java/com/fitness/configuration/WorkoutSecurity.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    public boolean hasWorkoutId(Authentication authentication, UUID exerciserUuid, Long workoutId) throws NotFoundException {
+
+        hasExerciserId(authentication, exerciserUuid);
+
+        Exerciser exerciser = exerciserService.findByUuid(exerciserUuid);
+
+        Workout workout = workoutRepository.findById(workoutId);
+
```
Comment (MarkHychka) [inline src/main/java/com/fitness/configuration/WorkoutSecurity.java]: why do you add empty line after every line of code?

File: src/main/java/com/fitness/configuration/WorkoutSecurity.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    ExerciserService exerciserService;
+
+    public boolean hasExerciserId(Authentication authentication, UUID exerciserUuid) {
+
+        UUID uuid = exerciserRepository.findByEmail(authentication.getPrincipal().toString()).get(0).getUuid();
+
+        return uuid.equals(exerciserUuid);
+    }
+
+    public boolean hasWorkoutId(Authentication authentication, UUID exerciserUuid, Long workoutId) throws NotFoundException {
```
Comment (MarkHychka) [inline src/main/java/com/fitness/configuration/WorkoutSecurity.java]: hasWorkoutId -> hasPermission

You don't need `exerciserUuid` here. You can find an exerciser by email from authentication and check his id with exerciserId from workout

File: src/main/java/com/fitness/configuration/WorkoutSecurity.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    @Autowired
+    ExerciserRepository exerciserRepository;
+
+    @Autowired
+    WorkoutRepository workoutRepository;
+
+    @Autowired
+    ExerciserService exerciserService;
+
+    public boolean hasExerciserId(Authentication authentication, UUID exerciserUuid) {
```
Comment (MarkHychka) [inline src/main/java/com/fitness/configuration/WorkoutSecurity.java]: Create class ExerciserSecurity and move this method in this class.
hasExerciserId -> hasPermission

File: src/main/java/com/fitness/controller/WorkoutController.java [hunk 7]
```diff
@@ -29,26 +29,26 @@
     private WorkoutService workoutService;
 
     @PostMapping(value = "/exerciser/{exerciserUuid}/workout")
-    @PreAuthorize("hasRole('ROLE_EXERCISER')")
+    @PreAuthorize("hasRole('ROLE_EXERCISER') and @workoutSecurity.hasExerciserId(authentication,#exerciserUuid)")
```
Comment (MarkHychka) [inline src/main/java/com/fitness/controller/WorkoutController.java]: You secured WorkoutController but what about ExerciserController and ClubController?

File: src/main/java/com/fitness/configuration/WorkoutSecurity.java (hunk 5 above)
Comment (EdikFotul) [reply src/main/java/com/fitness/configuration/WorkoutSecurity.java]: Here i need `exerciseUuid` cuz i check in same time is correct `exerciseUuid` and `workoutId` used for request

Comment (MarkHychka) [reply src/main/java/com/fitness/configuration/WorkoutSecurity.java]: hasWorkoutId -> hasPermission

**Pull request merged**